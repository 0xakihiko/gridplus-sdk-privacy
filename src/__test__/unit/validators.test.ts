import {
  validateAddKvRequest,
  validateConnectRequest,
  validateFetchActiveWallet,
  validateGetAddressesRequest,
  validateGetKvRequest,
  validateRemoveKvRequest,
  validateSignRequest,
} from '../../functions';
import {
  buildGetAddressesObject,
  buildValidateConnectObject,
  buildValidateRequestObject,
} from '../utils/builders';

describe('validators', () => {
  describe('connect', () => {
    test('should successfully validate', () => {
      const connectBundle = buildValidateConnectObject({});
      const validConnectRequest = validateConnectRequest(connectBundle);
      expect(validConnectRequest.deviceId).toMatchSnapshot();
    });

    test('should throw errors on validation failure', () => {
      const connectBundle = buildValidateConnectObject({ baseUrl: '' });
      expect(() => validateConnectRequest(connectBundle)).toThrowError();
    });
  });

  describe('getAddresses', () => {
    test('should successfully validate', () => {
      const getAddressesBundle = buildGetAddressesObject({});
      expect(validateGetAddressesRequest(getAddressesBundle)).toMatchSnapshot();
    });

    test('should throw errors on validation failure', () => {
      const getAddressesBundle = buildGetAddressesObject({ url: '' });
      expect(() =>
        validateGetAddressesRequest(getAddressesBundle),
      ).toThrowError();
    });
  });

  describe('KvRecords', () => {
    describe('addKvRecords', () => {
      test('should successfully validate', () => {
        const validateAddKvBundle: any = buildValidateRequestObject({
          records: { key: 'value' },
        });
        expect(validateAddKvRequest(validateAddKvBundle)).toMatchSnapshot();
      });

      test('should throw errors on validation failure', () => {
        const validateAddKvBundle: any = buildValidateRequestObject({});
        expect(() => validateAddKvRequest(validateAddKvBundle)).toThrowError();
      });
    });

    describe('getKvRecords', () => {
      test('should successfully validate', () => {
        const validateGetKvBundle: any = buildValidateRequestObject({
          n: 1,
          type: 1,
          start: 0,
        });
        expect(validateGetKvRequest(validateGetKvBundle)).toMatchSnapshot();
      });

      test('should throw errors on validation failure', () => {
        const validateGetKvBundle: any = buildValidateRequestObject({ n: 0 });
        expect(() => validateGetKvRequest(validateGetKvBundle)).toThrowError();
      });
    });

    describe('removeKvRecords', () => {
      test('should successfully validate', () => {
        const validateRemoveKvBundle: any = buildValidateRequestObject({
          ids: [1],
          type: 1,
        });
        expect(
          validateRemoveKvRequest(validateRemoveKvBundle),
        ).toMatchSnapshot();
      });

      test('should throw errors on validation failure', () => {
        const validateRemoveKvBundle: any = buildValidateRequestObject({});
        expect(() =>
          validateRemoveKvRequest(validateRemoveKvBundle),
        ).toThrowError();
      });
    });
  });

  describe('fetchActiveWallet', () => {
    test('should successfully validate', () => {
      const validateFetchActiveWalletBundle: any = buildValidateRequestObject(
        {},
      );
      expect(
        validateFetchActiveWallet(validateFetchActiveWalletBundle),
      ).toMatchSnapshot();
    });

    test('should throw errors on validation failure', () => {
      const validateFetchActiveWalletBundle: any = { url: '' };
      expect(() =>
        validateFetchActiveWallet(validateFetchActiveWalletBundle),
      ).toThrowError();
    });
  });

  describe('sign', () => {
    test('should successfully validate', () => {
      const validateSignRequestBundle: any = buildValidateRequestObject({
        wallet: 'wallet',
      });
      expect(validateSignRequest(validateSignRequestBundle)).toMatchSnapshot();
    });

    test('should throw errors on validation failure', () => {
      const validateSignRequestBundle: any = buildValidateRequestObject({});
      expect(() =>
        validateSignRequest(validateSignRequestBundle),
      ).toThrowError();
    });
  });
});
