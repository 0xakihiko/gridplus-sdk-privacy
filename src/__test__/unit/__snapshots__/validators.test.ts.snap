// Vitest Snapshot v1

exports[`validators > KvRecords > addKvRecords > should successfully validate 1`] = `
{
  "fwConstants": {
    "kvActionsAllowed": true,
  },
  "sharedSecret": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "url": "test.com",
  "validRecords": {
    "key": "value",
  },
}
`;

exports[`validators > KvRecords > getKvRecords > should successfully validate 1`] = `
{
  "fwConstants": {
    "kvActionsAllowed": true,
  },
  "n": 1,
  "sharedSecret": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "start": 0,
  "type": 1,
  "url": "test.com",
}
`;

exports[`validators > KvRecords > removeKvRecords > should successfully validate 1`] = `
{
  "fwConstants": {
    "kvActionsAllowed": true,
  },
  "ids": [
    1,
  ],
  "sharedSecret": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "type": 1,
  "url": "test.com",
}
`;

exports[`validators > connect > should successfully validate 1`] = `"test"`;

exports[`validators > fetchActiveWallet > should successfully validate 1`] = `
{
  "sharedSecret": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "url": "test.com",
}
`;

exports[`validators > getAddresses > should successfully validate 1`] = `
{
  "fwVersion": {
    "data": [
      0,
      12,
      0,
    ],
    "type": "Buffer",
  },
  "sharedSecret": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "url": "asdf",
  "wallet": {
    "capabilities": 1,
    "external": true,
    "name": {
      "data": [
        116,
        101,
        115,
        116,
      ],
      "type": "Buffer",
    },
    "uid": {
      "data": [
        116,
        101,
        115,
        116,
      ],
      "type": "Buffer",
    },
  },
}
`;

exports[`validators > sign > should successfully validate 1`] = `
{
  "fwConstants": {
    "kvActionsAllowed": true,
  },
  "sharedSecret": {
    "data": [
      116,
      101,
      115,
      116,
    ],
    "type": "Buffer",
  },
  "url": "test.com",
  "wallet": "wallet",
}
`;
